pkg/common/data_store.go

```diff
@@ -23,8 +25,33 @@ func OffloadLiteralMap(ctx context.Context, storageClient *storage.DataStore, li
 	if err != nil {
 		return "", errors.NewFlyteAdminErrorf(codes.Internal, "Failed to construct data reference for [%+v] with err: %v", nestedKeys, err)
 	}
-	if err := storageClient.WriteProtobuf(ctx, uri, storage.Options{}, literalMap); err != nil {
+
+	err = Retry(5, async.RetryDelay, func() error {
+		err = storageClient.WriteProtobuf(ctx, uri, storage.Options{}, literalMap)
+		return err
+	})
+
+	if err != nil {
 		return "", errors.NewFlyteAdminErrorf(codes.Internal, "Failed to write protobuf for [%+v] with err: %v", nestedKeys, err)
 	}
+
 	return uri, nil
 }
+
+func Retry(attempts int, delay time.Duration, f func() error) error {
+	var err error
+	for attempt := 0; attempt <= attempts; attempt++ {
+		err = f()
+		if err == nil {
+			return nil
+		}
+		if e, ok := err.(*googleapi.Error); ok && e.Code == 409 {
```

Comment:
A bit hard to reproduce this as my local version does not connect to GCP and even if I am not sure I could get this error to reappear.  I used this https://github.com/graymeta/stow/blob/973a61f346d598a566affb53c4698764a67df164/google/location.go#L38-L42 to understand how to handle it